### PR TITLE
python calling rust example

### PR DESCRIPTION
**Pull request by mfarrugi**
This is not well supported by python rules, but sorta works. https://github.com/bazelbuild/bazel/issues/1475 outlines the hacky setup.

Might also want to create a pyo3 or similar example outside of this repo.

### CODE REVIEW

**Comment by acmcarther** [comment]
I can't unblock this for CI. Could I bother you to push a commit (perhaps removing "debug stuff", unless you think you'll need it to debug OSX)?

I think the ci pipeline forgot about this verification request and the link is now 404ing.

**Comment by mfarrugi** [comment]
I don't think it's right referring to the mangled solib path, but don't have a good idea of what The Right Way is yet. You just need any commit to re-trigger CI?

**Comment by acmcarther** [comment]
Oops, didn't submit my comment!

Yeah I just need a commit so that the CI sees it.

File: examples/ffi/python_calling_rust/py_library_from_rust.bzl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        # TODO: Was there a template rule somewhere?
+        cmd = """
+        cat << EOM > $(OUTS)
+
+from __future__ import print_function
+
+import os
+from ctypes import cdll
+
+# TODO: This is a hack to compensate for rootpath not giving us the _solib_* path...
```
Comment (mfarrugi) [inline examples/ffi/python_calling_rust/py_library_from_rust.bzl]: I cannot quite put my finger on how hacky this is.. let me know what ya'll think.

Comment (mfarrugi) [reply examples/ffi/python_calling_rust/py_library_from_rust.bzl]: This ultimately searches all of runfiles, so unless the design of collecting shared libraries in one location changes dramatically, it should be safe. That said, searching should really not be necessary and I'm not sure if it's a deficiency in python/c++ rules or my lack of knowledge here.

**Comment by mfarrugi** [comment]
I did this mostly to see how difficult / complicated setting up the integration was, I think it's reasonably straightforward (and would be dead simple if the $(rootpath)  gave the path we wanted).

I think the best case scenario for this sort of integration is for python to have the capacity to take a header and a shared library and generate all of the boilerplate, and for rust to be able to produce the shared library and header (with cbindgen). IIUC this is roughly how [milksnake](https://github.com/getsentry/milksnake) works, as opposed to directly creating a python native extension. The API would look roughly the same as the current example, however would work for functions that aren't `i32` in and `i32` out.

In general I don't have an immediate use for this, but I think it's nice to say "look, it's easy to use rust with xxx". It is okay with me for this to continue to sit in a pull request (it'll be a year old soon!), but I don't mind providing a similar level of guidance / support as with the rest of the rules.

**Comment by aiuto** [comment]
I would rather not have it sit in a PR forever. if it works and we can put it in CI, then you should dust it off to commit it. If it doesn't really work, how about turning it into examples/ffi/python_calling_rust/readme.md. There you can describe the approach and how a more disciplined solution might work. A PR is essentially undiscoverable.

**Comment by mfarrugi** [comment]
Will close this for now, should be just as easy to find.

Would make sense to revisit when bazel-python has support for native extensions.

**Pull request closed without merging**